include/pingcap/kv/LockResolver.h

```diff
@@ -23,7 +23,24 @@ struct TxnStatus
     uint64_t commit_ts = 0;
     ::kvrpcpb::Action action;
     std::optional<::kvrpcpb::LockInfo> primary_lock;
-    bool isCommited() { return ttl == 0 && commit_ts > 0; }
+    bool isCommitted() { return ttl == 0 && commit_ts > 0; }
+
+    bool isCacheable()
+    {
+        if (isCommitted())
+        {
+            return true;
+        }
+        if (ttl == 0)
+        {
+            if (action == kvrpcpb::Action::NoAction || action == kvrpcpb::Action::LockNotExistRollback
```

Comment:
try switch case ?

Reply:
This can be expressed by one IF so I think we don't need to use SWITCH.

Reply:
actually switch case is more efficient on runtime.

Reply:
Maybe for such simple case, compiler itself will make optimization.